lib/ruby_ui/combobox/combobox_controller.js

```diff
@@ -1,201 +1,249 @@
 import { Controller } from "@hotwired/stimulus";
-import { computePosition, autoUpdate, offset } from "@floating-ui/dom";
-
-export const POPOVER_OPENED = "ruby-ui--combobox#popoverOpened";
 
+// Connects to data-controller="ruby-ui--combobox"
 export default class extends Controller {
-  static targets = [
-    "input",
-    "trigger",
-    "value",
-    "content",
-    "search",
-    "list",
-    "item",
-  ];
-  static values = { open: Boolean };
-  static outlets = ["ruby-ui--combobox-item", "ruby-ui--combobox-content"];
-
-  constructor(...args) {
-    super(...args);
-    this.cleanup;
+  static values = {
+    multiple: Boolean, // controls if select input accepts multiple selected options or not
+    term: String // text used on multiple combobox to indicate how many items are selected, eg. `4 items`
   }
 
+  static targets = [
+    "dialog", // dialog that shows combobox options
+    "datalistOption", // visual options inside dialog
+    "emptyState", // element displayed when the search doesn't return results
+    "searchInput", // search input used for filtering options
+    "select", // hidden select that control combobox value
+    "trigger", // button that opens dialog
+    "triggerContent" // button content used to display selected values
+  ]
+
+  selectedOptionIndex = null
+
   connect() {
-    this.#setFloatingElement();
-    this.#generateItemsIds();
+    this.initSelect()
+    this.updateTriggerContent()
   }
 
-  disconnect() {
-    this.cleanup();
+  // Init select using datalist options
+  initSelect() {
+    if (this.multipleValue) {
+      this.selectTarget.setAttribute("multiple", true)
+    }
+
+    this.datalistOptionTargets.forEach((datalistOption) => {
+      const selectOption = document.createElement("option")
```

Comment:
I was wondering if this would work if the combobox components were loaded using Turbo/async. Relying on JavaScript to create elements might be risky.